src/main/java/bio/terra/service/dataset/flight/ingest/CombinedFileIngestOptionalStep.java

```diff
@@ -0,0 +1,16 @@
+package bio.terra.service.dataset.flight.ingest;
+
+import bio.terra.service.job.OptionalStep;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+
+public class CombinedFileIngestOptionalStep extends OptionalStep {
+  public CombinedFileIngestOptionalStep(Step step) {
+    super(step);
+  }
```

Comment:
Ok - I don't know if I'm totally understanding how this works.  Do you have to set up a class that extends OptionalStep for each isEnabled() conditional? 
So, for the the case of SnapshotDelete, would that mean we'd have to add 6 additional classes for the 6 different predicates?

Reply:
Yes, each conditional would have a different class. The idea is that 

- there are steps that share the same conditional
- a step can be made optional at any point by wrapping it in `OptionalStep`. The step doesn't need to be changed to be made optional. Right now whenever a step is made optional / not optional its base class needs to be changed and its do/undo methods must be renamed.

Creating classes also lets us move the methods out of a Utils class and into a class that can "own" them. In many cases a Utils class is a sign that code isn't yet OO in some way. (In other cases it's because the object whose behavior we want to enhance isn't owned by us so we can't add methods to it.)